### ISSUE TITLE
Code sample for date range picker results in 508 non-compliance

### ISSUE THREAD

**Issue by bmcorum** [uswds/uswds#4347]
## Description

Noticed that both the component preview and component code section results in a 508 compliance error due to the misusage of the aria-describedby attribute and a missing aria-labeledby attribute.

The input tag should be
`<input
        class="usa-input"
        id="event-date-start"
        name="event-date-start"
        type="text"
        aria-labeledby="event-date-start-label"
        aria-describedby="event-date-start-hint"
      />`

## Steps to reproduce the issue

N/A

## Additional information [optional]
![image](https://user-images.githubusercontent.com/1165615/134227885-e8aef8b3-f24e-4893-84ca-122ba501be82.png)

**Comment by amyleadem** [comment]
Confirming that Wave does display a 'missing form label' error on this component.  

Looks as though the error is flagged on the hidden <input> elements with the class "usa-date-picker__internal-input". These elements do not currently have an associated label nor an aria-labelledby attribute.

As is, VoiceOver is able to provide correct description and label to the user on the visible ("usa-date-picker__external-input") input elements. However, the Wave docs recommend using proper accessibility attributes even on hidden elements. Reference: https://wave.webaim.org/help#hidden

I was able to remove the Wave error by adding 'aria-labelledby="event-date-start-label"' to the "usa-date-picker__internal-input" elements.

**Comment by bmcorum** [comment]
@brunerae the component will initialize correctly if the HTML input includes the correct usage of aria-describedby attribute and a missing aria-labeledby attribute. The only issue here was that the documentation and code sample are using the attributes incorrectly thus leading to the accessibility issue. I would be more than glad to submit a PR for this since it would only be a documentation and code sample update, however I could not for my best efforts locate the files used to produce the documentation on [https://designsystem.digital.gov/components/date-range-picker/](https://designsystem.digital.gov/components/date-range-picker/).

**Comment by mejiaj** [comment]
@bmcorum thanks for your interest in contributing! I see the issue in both [date range picker](https://designsystem.digital.gov/components/date-range-picker/) and [date picker](https://designsystem.digital.gov/components/date-picker/).

**Code**
In USWDS's codebase you can find the templates here:
- [Date range picker](https://github.com/uswds/uswds/blob/e0bbffa9819d99ed2c262e94e9d89f1635e566ac/src/components/date-range-picker/date-range-picker.njk#L6)
- [Date picker](https://github.com/uswds/uswds/blob/e0bbffa9819d99ed2c262e94e9d89f1635e566ac/src/components/date-picker/date-picker.njk#L16)

**Guidance**
- [Date range picker](https://github.com/uswds/uswds-site/blob/6d691654cd051c58a4711c27df900c2311fee86c/_components/date-range-picker/guidance/accessibility.md)
- [Date picker](https://github.com/uswds/uswds-site/blob/6d691654cd051c58a4711c27df900c2311fee86c/_components/date-picker/guidance/accessibility.md)

**Comment by mejiaj** [comment]
The [form guidance](https://designsystem.digital.gov/components/form/) says:

>VoiceOver on OS X currently does not support aria-describedby. Use aria-labelledby instead, and include all related fields, including, labels, legend, and hint text

But there does seem to be [partial support](https://a11ysupport.io/tech/aria/aria-describedby_attribute) now based on a11ysupport.io. We just need to find the best way to communicate this partial support.

**Comment by mejiaj** [comment]
**Code PR:** https://github.com/uswds/uswds/pull/4414
**Guidance PR:** https://github.com/uswds/uswds-site/pull/1363
